src/app-host/app-host.js

```diff
@@ -5,41 +5,85 @@ var Messages = require('messages');
 var telemetry = require('telemetry-helper');
 
 var cordova;
-var oldExec;
 var socket = io();
 var nextExecCacheIndex = 0;
 var execCache = {};
 var pluginHandlers = {};
 var serviceToPluginMap = {};
 
 function setCordova(originalCordova) {
+    var channel;
+
     if (cordova) {
         return;
     }
 
     cordova = originalCordova;
 
-    oldExec = cordova.require('cordova/exec');
     cordova.define.remove('cordova/exec');
     cordova.define('cordova/exec', function (require, exports, module) {
         module.exports = exec;
     });
 
-    // android platform has its own specific initialization
-    // so to emulate it, we need to fake init function
-    if (cordova.platformId === 'android') {
-        exec.init = function () {
-            cordova.require('cordova/channel').onNativeReady.fire();
-        };
-    }
+    channel = cordova.require('cordova/channel');
+
+    // define our own channel to delay the initialization until sim-host tells
+    // us everything's ready (fired in 'start' event handler).
+    channel.createSticky('onCordovaSimulateReady');
+    channel.waitForInitialization('onCordovaSimulateReady');
+
+    socket.once('init', function () {
+        // sim-host is ready, register exec handlers, fire onNativeReady and send
+        // the list of plugins
+        socket.on('exec-success', function (data) {
+            console.log('exec-success:');
+            console.log(data);
+            var execCacheInfo = execCache[data.index];
+            if (execCacheInfo.success) {
+                execCacheInfo.success(data.result);
+            }
+        });
+
+        socket.on('exec-failure', function (data) {
+            console.log('exec-failure:');
+            console.log(data);
+            var execCacheInfo = execCache[data.index];
+            if (execCacheInfo.fail) {
+                execCacheInfo.fail(data.error);
+            }
+        });
+
+        socket.on('start-live-reload', function () {
+            livereload.start(socket);
+        });
+
+        socket.on('init-telemetry', function (data) {
+            telemetry.init(socket);
+        });
+
+        socket.on('init-xhr-proxy', function (data) {
+            require('xhr-proxy').init(); 
+        });
+
+        channel.onNativeReady.fire();
```

Comment:
Since plugin's simulation code acts as the native side of the plugin, those should be initialized and ready before firing `onNativeRead`, what do you think @busykai ?